framework/src/media/voice/HardwareEndPointDetector.cpp

```diff
@@ -16,22 +16,115 @@
  *
  ******************************************************************/
 #include "HardwareEndPointDetector.h"
+#include "../audio/audio_manager.h"
 
 namespace media {
 namespace voice {
 
+HardwareEndPointDetector::HardwareEndPointDetector(int card, int device)
+	: mCard(card)
+	, mDevice(device)
+{
+
+}
+
 bool HardwareEndPointDetector::init(uint32_t samprate, uint8_t channels)
 {
+	audio_manager_result_t result;
+
+	result = register_stream_in_device_process_type(
+		mCard, mDevice,
+		AUDIO_DEVICE_PROCESS_TYPE_SPEECH_DETECTOR,
+		AUDIO_DEVICE_SPEECH_DETECT_EPD);
+
+	if (result != AUDIO_MANAGER_SUCCESS) {
+		meddbg("Error: register_stream_in_device_process_type(%d, %d) failed!\n", mCard, mDevice);
+		return false;
+	}
+
+	result = register_stream_in_device_process_handler(
+		mCard, mDevice,
+		AUDIO_DEVICE_PROCESS_TYPE_SPEECH_DETECTOR);
+
+	if (result != AUDIO_MANAGER_SUCCESS) {
+		meddbg("Error: register_stream_in_device_process_handler(%d, %d) failed!\n", mCard, mDevice);
+		return false;
+	}
+
+	result = start_stream_in_device_process(mCard, mDevice);
+	if (result != AUDIO_MANAGER_SUCCESS) {
+		meddbg("Error: start_stream_in_device_process(%d, %d) failed!\n", mCard, mDevice);
+		return false;
+	}
+
+	pthread_t test;
+    pthread_attr_t attr;
```

Comment:
Need to check indentations in line 61 ~ 69.

Reply:
You're right. I'll fix